src/main/scala/io/iohk/ethereum/transactions/PendingTransactionsManager.scala

```diff
@@ -0,0 +1,49 @@
+package io.iohk.ethereum.transactions
+
+import akka.actor.{Actor, ActorRef, Props}
+import io.iohk.ethereum.domain.SignedTransaction
+import io.iohk.ethereum.network.PeerManagerActor
+import io.iohk.ethereum.network.PeerMessageBusActor.{MessageClassifier, MessageFromPeer, PeerSelector, Subscribe}
+import io.iohk.ethereum.network.p2p.messages.CommonMessages.SignedTransactions
+
+object PendingTransactionsManager {
+  def props(peerManager: ActorRef, peerMessageBus: ActorRef): Props =
+    Props(new PendingTransactionsManager(peerManager, peerMessageBus))
+
+  case class BroadcastTransaction(signedTransaction: SignedTransaction)
+
+  case object GetPendingTransactions
+  case class PendingTransactions(signedTransactions: Seq[SignedTransaction])
+
+  case class RemoveTransactions(signedTransactions: Seq[SignedTransaction])
+}
+
+class PendingTransactionsManager(peerManager: ActorRef, peerMessageBus: ActorRef) extends Actor {
+
+  import PendingTransactionsManager._
+
+  var pendingTransactions: Seq[SignedTransaction] = Nil
+
+  peerMessageBus ! Subscribe(MessageClassifier(Set(SignedTransactions.code), PeerSelector.AllPeers))
+
+  override def receive: Receive = {
+    case BroadcastTransaction(signedTransaction) =>
+      if (!pendingTransactions.contains(signedTransaction)) {
+        peerManager ! PeerManagerActor.BroadcastMessage(SignedTransactions(Seq(signedTransaction)))
+        pendingTransactions :+= signedTransaction
+      }
+
+    case GetPendingTransactions =>
+      sender() ! PendingTransactions(pendingTransactions)
+
+    case RemoveTransactions(signedTransactions) =>
```

Comment:
Should this be used to remove the transactions from processed blocks on regular sync? This may become complicated as we should handle when there is some blockchain reorganization and we should add txs from dropped block to the list of `pendingTransactions`, so we could create a separate task for this if needed.